data/service/service/client.go

```diff
@@ -37,6 +39,89 @@ func (c *Client) GetDataSet(ctx context.Context, id string) (*data.DataSet, erro
 	return repository.GetDataSet(ctx, id)
 }
 
+func (c *Client) GetSummary(ctx context.Context, id string) (*data.Summary, error) {
+	repository := c.dataStore.NewSummaryRepository()
+
+    summary, err := repository.GetSummary(ctx, id)
+
+    if err == mongo.ErrNoDocuments {
+        dataRepository := c.dataStore.NewDataRepository()
+
+        // check to ensure the user has data
+        _, err := dataRepository.GetLastUpdated(ctx, id)
+        if err != nil {
+            return nil, err
+        }
+
+        // create empty summary to ensure calc in the future
+        summary.UserID = id
+        summary.LastUpdated = time.Time{}
+        repository.UpdateSummary(ctx, summary)
+    }
+    return summary, nil
+}
+
+func (c *Client) UpdateSummary(ctx context.Context, id string) (*data.Summary, error) {
```

Comment:
Does this need to consider race conditions, given that it needs the original summary for rolling calc and the fetch/compute/update operation is not atomic?

Reply:
It should already be race safe, as it uses the original summary's lastData as it's lower bound, and then dynamically calculates the weighting after finding the latest data, then queries by that specific range.